ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/RPad.java

```diff
@@ -49,4 +52,41 @@ public String rpad(
     sb.setLength(targetLen);
     return sb.toString();
   }
+
+  @Udf
+  public ByteBuffer rpad(
+      @UdfParameter(description = "Bytes to be padded") final ByteBuffer input,
+      @UdfParameter(description = "Target length") final Integer targetLen,
+      @UdfParameter(description = "Padding bytes") final ByteBuffer padding) {
+
+    if (input == null) {
+      return null;
+    }
+    if (padding == null
+        || BytesUtils.getByteArray(padding).length == 0
+        || targetLen == null
+        || targetLen < 0) {
+      return null;
+    }
+
+    final byte[] start = BytesUtils.getByteArray(input);
+
+    if (start.length > targetLen) {
+      return ByteBuffer.wrap(Arrays.copyOfRange(start, 0, targetLen));
+    }
+
+    final byte[] padded = new byte[targetLen];
+    final byte[] paddingArray = BytesUtils.getByteArray(padding);
+
+    for (int i = 0; i < targetLen; i++) {
+      if (i < start.length) {
+        padded[i] = start[i];
+      } else {
+        final int paddingIndex = (i - start.length) % paddingArray.length;
+        padded[i] = paddingArray[paddingIndex];
+      }
+
+    }
```

Comment:
You could start copying the initial bytes by using `Sytem.arraycopy`, and then just adding the padding bytes.

```
// This copies all bytes from `start` to `padded`.
// See https://www.tutorialspoint.com/java/lang/system_arraycopy.htm
System.arraycopy(start, 0, padded, 0, start.length);
```

Reply:
Good idea, this is cleaner.